tests/unit/core/server/testServer.js

```diff
@@ -157,20 +157,35 @@ describe(__filename, () => {
         .set('cookie', `${defaultConfig.get('cookieName')}="${token}"`)
         .end();
 
-      const { api, user } = store.getState();
+      const {
+        api,
+        user,
+        reduxAsyncConnect: reduxAsyncConnectState,
+      } = store.getState();
 
       expect(response.statusCode).toEqual(200);
       expect(api.token).toEqual(token);
       expect(user.id).toEqual(42);
       expect(user.username).toEqual('babar');
       mockUserApi.verify();
+
+      const { loadState } = reduxAsyncConnectState;
+      expect(loadState).toEqual({});
     });
 
     it('returns a 500 error page when retrieving the user profile fails', async () => {
+      const errorMessage = 'example of an API error';
+      const error = Promise.reject(new Error(errorMessage));
```

Comment:
Ugh, this is a confusing workaround. 

I think it would be helpful to make a test helper for this that looks like:

```js
const errorResult = rejectedPromise(new Error(errorMessage));
```

This would allow us to put the comments in the helper function itself.

We use `Promise.reject()` all over the test suite so I don't understand why only this test needed updating. Is it something to do with how this promise runs in a separate process? I.E. the test server promise.

Reply:
A helper works for me.

I think the problem comes from the server-side rendering indeed. We boot a real Express app and I think node or jest loses track of the promise and thinks it is not handled (that is what I understand from this blog post: http://thecodebarbarian.com/unhandled-promise-rejections-in-node.js.html, especially the section about to sinon and stubs).

I have attached `.catch()` in many different places and tried a few other `try`/`catch` combinations. I am not able to catch this "unhandled rejection warning".

Not sure if we can do anything else, because it seems we have to hook into `process.on('unhandledRejection', () => {})`. That is also what Jest does since version 21 actually. If I comment the Jest code related to that, no more issue... I tried to catch the error in the sagas, but that was not working either.

I don't think this is a perfect solution though. One more thing, I used node 8 too and could not generate unhandled promise warnings using the app for real, but the test suite was failing like on Travis.